src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -227,16 +235,33 @@ await InvokeAsync(() =>
 
         void SetSelectedResourceOption(ResourceViewModel resource)
         {
-            Debug.Assert(_resources is not null);
-
-            PageViewModel.SelectedOption = _resources.Single(option => option.Id?.Type is not OtlpApplicationType.ResourceGrouping && string.Equals(ResourceName, option.Id?.InstanceId, StringComparison.Ordinal));
+            PageViewModel.SelectedOption = GetSelectedOption();
             PageViewModel.SelectedResource = resource;
 
             Logger.LogDebug("Selected console resource from name {ResourceName}.", ResourceName);
             loadingTcs.TrySetResult();
         }
     }
 
+    private SelectViewModel<ResourceTypeDetails> GetSelectedOption()
+    {
+        Debug.Assert(_resources is not null);
+        return _resources.SingleOrDefault(
+            option =>
+            {
+                var typeDetails = option.Id;
+                if (typeDetails is null || typeDetails.Type is OtlpApplicationType.ResourceGrouping)
+                {
+                    return false;
+                }
+
+                return string.Equals(ResourceName, typeDetails.Type is OtlpApplicationType.Instance
+                    ? option.Id?.InstanceId
+                    : option.Name, StringComparisons.ResourceName);
```

Comment:
```suggestion
                var optionName = typeDetails.Type is OtlpApplicationType.Instance
                    ? option.Id?.InstanceId
                    : option.Name;
                return string.Equals(ResourceName, optionName, StringComparisons.ResourceName);
```

Reply:
And add a comment about the difference between instance id and name in this situation.